core/commands/store/upload/helper/guard.go

```diff
@@ -0,0 +1,343 @@
+	}, nil
+}
+
+func getGuardAndEscrowPid(configuration *config.Config) (peer.ID, peer.ID, error) {
+	escrowPubKeys := configuration.Services.EscrowPubKeys
+	if len(escrowPubKeys) == 0 {
+		return "", "", fmt.Errorf("missing escrow public key in config")
+	}
+	guardPubKeys := configuration.Services.GuardPubKeys
+	if len(guardPubKeys) == 0 {
+		return "", "", fmt.Errorf("missing guard public key in config")
+	}
+	escrowPid, err := pidFromString(escrowPubKeys[0])
+	if err != nil {
+		log.Error("parse escrow config failed", escrowPubKeys[0])
+		return "", "", err
+	}
+	guardPid, err := pidFromString(guardPubKeys[0])
+	if err != nil {
+		log.Error("parse guard config failed", guardPubKeys[1])
+		return "", "", err
+	}
+	return guardPid, escrowPid, err
+}
+
+func pidFromString(key string) (peer.ID, error) {
+	pubKey, err := escrow.ConvertPubKeyFromString(key)
+	if err != nil {
+		return "", err
+	}
+	return peer.IDFromPublicKey(pubKey)
+}
+
+const (
+	guardTimeout = 360 * time.Second
+)
+
+func PrepAndUploadFileMeta(ctx context.Context, contracts []*guardPb.Contract, payinRes *escrowPb.SignedPayinResult,
+	payerPriKey ic.PrivKey, configuration *config.Config, renterId string, fileHash string) (*guardPb.FileStoreStatus,
+	error) {
+	sig := payinRes.EscrowSignature
+	for _, guardContract := range contracts {
+		guardContract.EscrowSignature = sig
+		guardContract.EscrowSignedTime = payinRes.Result.EscrowSignedTime
+		guardContract.LastModifyTime = time.Now()
+	}
+
+	fileStatus, err := NewFileStatus(contracts, configuration, renterId, fileHash)
+	if err != nil {
+		return nil, err
+	}
+
+	sign, err := crypto.Sign(payerPriKey, &fileStatus.FileStoreMeta)
+	if err != nil {
+		return nil, err
+	}
+	if fileStatus.PreparerPid == fileStatus.RenterPid {
+		fileStatus.RenterSignature = sign
+	} else {
+		fileStatus.RenterSignature = sign
+		fileStatus.PreparerSignature = sign
+	}
+
+	err = submitFileStatus(ctx, configuration, fileStatus)
+	if err != nil {
+		return nil, err
+	}
+
+	return fileStatus, nil
+}
+
+func NewFileStatus(contracts []*guardPb.Contract, configuration *config.Config,
+	renterId string, fileHash string) (*guardPb.FileStoreStatus, error) {
+	guardPid, escrowPid, err := getGuardAndEscrowPid(configuration)
+	if err != nil {
+		return nil, err
+	}
+	var (
+		rentStart   time.Time
+		rentEnd     time.Time
+		preparerPid = renterId
+		renterPid   = renterId
+		rentalState = guardPb.FileStoreStatus_NEW
+	)
+	if len(contracts) > 0 {
+		rentStart = contracts[0].RentStart
+		rentEnd = contracts[0].RentEnd
+		preparerPid = contracts[0].PreparerPid
+		renterPid = contracts[0].RenterPid
+		if contracts[0].PreparerPid != contracts[0].RenterPid {
+			rentalState = guardPb.FileStoreStatus_PARTIAL_NEW
+		}
+	}
+
+	fileStoreMeta := guardPb.FileStoreMeta{
+		RenterPid:        renterPid,
+		FileHash:         fileHash,  //TODO need to check
+		FileSize:         10000,     //TODO need to revise later
```

Comment:
this looks like a regression - we fixed this already.... need to double check with master